ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1210,16 +1210,24 @@ func (r *RayClusterReconciler) calculateStatus(ctx context.Context, instance *ra
 			meta.SetStatusCondition(&newInstance.Status.Conditions, headPodReadyCondition)
 		}
 
-		if meta.FindStatusCondition(newInstance.Status.Conditions, string(rayv1.RayClusterProvisioned)) == nil {
+		if meta.FindStatusCondition(newInstance.Status.Conditions, string(rayv1.RayClusterProvisioned)) == nil ||
+			meta.IsStatusConditionFalse(newInstance.Status.Conditions, string(rayv1.RayClusterProvisioned)) {
```

Comment:
How about change these lines to `if !meta.IsStatusConditionTrue(newInstance.Status.Conditions, string(rayv1.RayClusterProvisioned)`? It looks simpler and has the same meaning.

ref: https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta@v0.30.2#IsStatusConditionTrue

Reply:
Good catch! Will update